File: backend-app/src/main/java/tom/projects/monkeylog/service/UserService.java
package tom.projects.monkeylog.service;

import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;
import org.springframework.web.server.ResponseStatusException;
import tom.projects.monkeylog.model.user.User;
import tom.projects.monkeylog.repository.user.UserRepository;
import tom.projects.monkeylog.security.AuthenticatedUser;

@Service
@RequiredArgsConstructor
public class UserService {
    private final UserRepository userRepository;

    public User get() {
        return userRepository.findById(AuthenticatedUser.getId())
                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found"));
    }
}

File: backendkt/src/main/kotlin/tom/com/monkeylog/service/UserService.kt
package tom.com.monkeylog.service

import org.springframework.http.HttpStatus
import org.springframework.stereotype.Service
import org.springframework.web.server.ResponseStatusException
import tom.com.monkeylog.model.user.User
import tom.com.monkeylog.repository.user.UserRepository
import tom.com.monkeylog.security.AuthenticatedUser

@Service
class UserService(val userRepository: UserRepository) {
    fun getUser(): User {
        return userRepository.findById(AuthenticatedUser.id)
            .orElseThrow { ResponseStatusException(HttpStatus.NOT_FOUND, "User not found") }
    }
}
